services/introspection/local.go

```diff
@@ -95,7 +94,12 @@ func (l *Local) Plugins(ctx context.Context, req *api.PluginsRequest, _ ...grpc.
 func (l *Local) getPlugins() []api.Plugin {
 	l.mu.Lock()
 	defer l.mu.Unlock()
-	return l.plugins
+	plugins := l.plugins.GetAll()
+	if l.pluginCache == nil || len(plugins) != l.numCached {
+		l.pluginCache = pluginsToPB(plugins)
+		l.numCached = len(plugins)
```

Comment:
`l.numCached` should be equal to `len(pluginCache)`, and it doesn't look like `numCached` is needed.
```go
if l.pluginCache == nil || len(plugins) != len(l.pluginCache) {
    l.pluginCache = pluginsToPB(plugins)
}
```
> Unless one needs to consider that `pluginsToPB` will filter some plugins, in which case `len(pluginCache)` and `numCached` are not equal.

Reply:
Yeah, you are right, they should be equal since `pluginsToPB` doesn't seem to do any filtering. Updated